cv_next/app/about/components/aboutLayout.tsx

```diff
@@ -1,17 +1,59 @@
+"use client";
 import Link from "next/link";
 import Image from "next/image";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
 import { buttonVariants } from "@/components/ui/button";
-import { heroHeader, routes, UI_Location } from "@/lib/definitions";
 import { cn } from "@/lib/utils";
+import {
+  Visible_Error_Messages,
+  heroHeader,
+  routes,
+  UI_Location,
+} from "@/lib/definitions";
+import PopupWrapper from "@/components/ui/popupWrapper";
 import { LoginButtons } from "./loginButtons";
 
 export const AboutLayout = () => {
+  const searchparams = useSearchParams();
+  const [errorMsg, setErrorMsg] = useState<string | null>();
+  const [errorDescription, setErrorDescription] = useState<string | null>();
+  const router = useRouter();
+
+  useEffect(() => {
+    const error = searchparams.get("error");
+    if (error == Visible_Error_Messages.InactiveUser.keyword) {
+      setErrorMsg(Visible_Error_Messages.InactiveUser.title);
+      setErrorDescription(Visible_Error_Messages.InactiveUser.description);
+    } else if (error != null) {
+      setErrorMsg(Visible_Error_Messages.DefaultError.title);
+      setErrorDescription(Visible_Error_Messages.DefaultError.description);
+    }
+  }, [searchparams]);
```

Comment:
I think now would be a good time to create an ErrorPopup component, or more preferably ErrorContext (with a ContextProvider) that'll catch an emitted error object (containing e.g. message, description, and callback).

Reply:
I think I added what you meant, lmk if more changes are needed